src/main/scala/org/broadinstitute/hail/variant/Variant.scala

```diff
@@ -166,6 +177,17 @@ object Variant {
         .map(s => AltAllele.fromRow(s)))
 }
 
+case class VariantSubGens(
+  contigGen: Gen[String] = Gen.identifier,
+  startGen: Gen[Int] = Gen.posInt,
+  nAlleleGen: Gen[Int] = Gen.frequency((5, Gen.const(2)), (1, Gen.choose(2, 10))),
+  allelesGen: (String, Int) => Gen[Array[String]] = (ref: String, nAlleles: Int) =>
+    Gen.distinctBuildableOfN[Array[String], String](
+    nAlleles,
+    Gen.frequency((10, genDNAString),
+      (1, Gen.const("*")))).filter(_(0) != "*")
```

Comment:
I did this because we want the reference to be a DNA string, while the alts can be whatever.